### PR TITLE
[FEATURE] Add match type to Match

### PR DESCRIPTION
**Pull request by psychomieze**

### CODE REVIEW

File: config/Matcher/ArrayDimensionMatcher.php [hunk 1]
```diff
@@ -106,7 +106,7 @@
     ],
     '$GLOBALS[\'TYPO3_CONF_VARS\'][\'EXT\'][\'extConf\']' => [
         'restFiles' => [
-            'Deprecation-82254-DeprecateGLOBALSTYPO3_CONF_VARSEXTextConf.rst',
+            'Deprecation-82254-DeprecateGLOBALSTYPO3_CONF_VARSEXTextConf.rst.twig',
```
Comment (ohader) [inline config/Matcher/ArrayDimensionMatcher.php]: .twig?

File: src/Scanner.php [hunk 2]
```diff
@@ -118,12 +128,16 @@ function (array $result) use ($matcher) {
 
         return array_map(
             function (array $result) {
+                if (!empty($result['restFiles'])) {
+                    $type = $this->inferTypeFromRestFiles($result['restFiles']);
```
Comment (ohader) [inline src/Scanner.php]: This could be empty `''`

File: src/Scanner.php [hunk 3]
```diff
@@ -118,12 +128,16 @@ function (array $result) use ($matcher) {
...
+                }
                 $match = new Match(
                     $result['matcher'],
                     $result['indicator'],
                     $result['subject'],
                     $result['message'],
-                    $result['line']
+                    $result['line'],
+                    $type ?? Match::TYPE_IMPORTANT
```
Comment (ohader) [inline src/Scanner.php]: Check for empty string as well

File: src/Scanner.php (hunk 2 above)
Comment (psychomieze) [reply src/Scanner.php]: done.

**Pull request merged**